cc-check.go

```diff
@@ -69,7 +69,11 @@ var requiredKernelModules = map[string]kernelModule{
 	"kvm_intel": {
 		desc: "Intel KVM",
 		parameters: map[string]string{
-			"nested":             "Y",
+			"nested": "Y",
+			// "VMX Unrestricted mode support". This is used
```

Comment:
I'd like to have @mcastelino and @anthonyzxu verify/clarify this. I think there is a more subtle set of interactions to do with `unrestricted_mode` (and if our QEMU supports software emulation of that if the hardware does not have the feature), and QEMU `disable-modern=true`, which changes the virtio PCIe IO/MMIO functionality. I expect there is a tight bond between using `modern MMIO` virtio PCIe devices and having `unrestricted_mode` supported or emulated in QEMU.

In summary, I think we can be clearer here - and maybe even note that if the feature is not present you can enable CC by modifying the hypervisor.args (QEMU command line - or maybe we add a config option for CC3.x runtime).

Reply:
pingety, ping! Any word-smiths out there? :smile:

Reply:
I think we need slightly more tech clarification here before we wordsmith it - I'm still not 100% clear on all the situations with `unrestricted_guest` and nested VMs.
We worked out the other bits - that we needed to revert to virtio 0.9 and we had to remove pmu or it tried to access some MSRs that did not exist in the VM - but, I cannot quite fathom where we ended up with the relationship of needing `unrestricted_guest` and nested VMs.
Over to @mcastelino and @sboeuf to clarify that I think (sorry guys - I know you are probably a bit fed up with that loop now)

Reply:
Graham,
I don't think we need to check unrestricted_guest.
Qemu-lite only run guest in protected paging mode.
Qemu-lite should work in machine which doesn't have unrestricted_guest feature.

Reply:
Hi @anthonyzxu - understood. But I believe we use this check as a heuristic to determine if the system is "new enough". Do we believe there could be users that will be trying to run on a system without this feature? If not, great, but if so, we need to find another heuristic to make `cc-check` more reliable.

It's worth pointing out that iff `unrestricted_guest` is not available, `cc-check` will only warn, not error. And of course, `cc-check` is not called by the runtime under a container manager so users will not be stopped from running if this feature is not available.

/cc @sameo, @grahamwhaley, @mcastelino.